pkg/repository/testworkflow/mongo.go

```diff
@@ -268,12 +269,42 @@ func (r *MongoRepository) GetExecutionsSummary(ctx context.Context, filter Filte
 	return
 }
 
-func (r *MongoRepository) Insert(ctx context.Context, result testkube.TestWorkflowExecution) (err error) {
+func (r *MongoRepository) Insert(ctx context.Context, result *testkube.TestWorkflowExecution) (err error) {
 	result.EscapeDots()
 	if result.Reports == nil {
 		result.Reports = []testkube.TestWorkflowReport{}
 	}
-	_, err = r.Coll.InsertOne(ctx, result)
+
+	wc := writeconcern.New(writeconcern.WMajority())
+	txnOptions := options.Transaction().SetWriteConcern(wc)
+	session, err := r.db.Client().StartSession()
+	if err != nil {
+		return err
+	}
+	defer session.EndSession(ctx)
+
+	_, err = session.WithTransaction(ctx, func(sessCtx mongo.SessionContext) (interface{}, error) {
```

Comment:
yes, transactions are supported only for replicaset and will not help, external lock with Mongo might work

Reply:
actualy, distributed lock is also not a silver bullet, if connection fails before the lock is removed, it will be necessary to wait till it is expired. so instead of simply rerunning the test and have a missed number, user will need to wait sometime and not sure, it's a better option for him. I would rather discuss, if we need it or not